README: update process description enhancement (#35)
osbuild/manifest-db: README.md

### Update workflow

The update workflow is made of 3 distinct steps:

#### First step, triggering the CI

The update workflow starts with a `Update DB` GitHub action that is manually
triggered. This action imports a subset of the manifests from osbuild-composer,
specifically those CI is able to build.

Importing manifests which cannot be built by CI will leave the
DB in a dirty state. For this reason they are excluded.

Once the import is done, a commit is created and pushed to [this GitLab repository](https://gitlab.com/redhat/services/products/image-builder/ci/manifest-db).

#### Second step, running the CI

> Using a trigger token the github action the CI start. Ci will use an other

Reviewer: 1. First sentence has a syntax mistake.  Not sure what it wants to say exactly.
2. CI in the second sentence isn't properly captialised.

workflow than the one used to run the regression testing.

> Each architecture and distribution listed in [.gitlab-ci.yml](https://github.com/osbuild/manifest-db/blob/main/.gitlab-ci.yml) will run in

Reviewer: I think this should be more something like
"A job for each architecture and distribution [...] will run in parallel"
because right now it's saying "each architecture and distro will run in parallel" and that doesn't really make sense.

parallel. Each job will generate and include the image-info for the manifests

> where possible and expose them as artifacts. As final step, a conclusion job

Reviewer: As a final step...

### Update workflow

The update workflow is made of 3 distinct steps:

#### First step, triggering the CI

> The update workflow starts with a `Update DB` GitHub action that is manually

Reviewer 2: ```suggestion
The update workflow starts with an `Update DB` GitHub action that is manually
```

triggered. This action imports a subset of the manifests from osbuild-composer,
specifically those CI is able to build.

Importing manifests which cannot be built by CI will leave the
DB in a dirty state. For this reason they are excluded.

Once the import is done, a commit is created and pushed to [this GitLab repository](https://gitlab.com/redhat/services/products/image-builder/ci/manifest-db).

#### Second step, running the CI

> Using a gitlab `trigger token` the github action starts the CI with a

Reviewer 2: ```suggestion
Using a GitLab `trigger token` the GitHub action starts the CI with a
```

### Update workflow

The update workflow is made of 3 distinct steps:

#### First step, triggering the CI

> The update workflow starts with a `Update DB` github action that is manually
> triggered. This action, imports a subset of the manifests from composer.
> Composed of the manifests the CI is able to build.

Reviewer 2: ```suggestion
The update workflow starts with a `Update DB` GitHub action that is manually
triggered. This action imports a subset of the manifests from osbuild-composer,
specifically those CI is able to build.
```

> The rationale behind is that importing a manifest CI can't build will leave the
> DB in a dirty state.

Reviewer 2: ```suggestion
Importing manifests which cannot be built by CI will leave the
DB in a dirty state. For this reason they are excluded.
```

> Once the importation is done, a commit is created and pushed to gitlab.

Reviewer 2: ```suggestion
Once the import is done, a commit is created and pushed to [this GitLab repository](https://gitlab.com/redhat/services/products/image-builder/ci/manifest-db).
```

> Gitlab URL: (https://gitlab.com/redhat/services/products/image-builder/ci/manifest-db).

Reviewer 2: ```suggestion
```

#### Second step, running the CI

Using a trigger token the github action the CI start. Ci will use an other
workflow than the one used to run the regression testing.

> Each architecture and distribution listed in .gitlab-ci.yml will run in

Reviewer 2: ```suggestion
Each architecture and distribution listed in [.gitlab-ci.yml](https://github.com/osbuild/manifest-db/blob/main/.gitlab-ci.yml) will run in
```

> parallel. Each job will generate and include as artifacts the image-info
> for each manifests it can. Once this is done, a conclusion job run. This job
> downloads every artifacts and include them into the commit that triggered the CI
> job.

Reviewer 2: ```suggestion
parallel. Each job will generate and include the image-info for the manifests
where possible and expose them as artifacts. As final step, a conclusion job
downloads every artifact and includes it in the commit that initially triggered
the CI job.
```

> #### Third step, opening a PR on github

Reviewer 2: ```suggestion
#### Third step, opening a PR on GitHub
```

> Then the commit is pushed onto github by Schutzbot and a PR is opened as a
> result.

Reviewer 2: ```suggestion
Finally the commit is pushed to GitHub by Schutzbot and a PR is opened.
```
